Mage.Sets/src/mage/cards/k/KillianInkDuelist.java

```diff
@@ -69,19 +69,21 @@ public KillianInkDuelistEffect copy() {
 
     @Override
     public boolean apply(Game game, Ability source, Ability abilityToModify) {
-        CardUtil.reduceCost(abilityToModify, 2);
+        // Always apply cost reduction for getPlayable and get real cost reduction once player has selected targets
+        // Bug #7762: https://github.com/magefree/mage/issues/7762
+        if (game.inCheckPlayableState() || CardUtil.getAllSelectedTargets(abilityToModify, game)
```

Comment:
1. Make each `inCheckPlayableState` conditional with docs like that:
![shot_210430_083403](https://user-images.githubusercontent.com/8344157/116648740-d5a8ed00-a98e-11eb-983b-59b16efbc9f0.png)

2. In check playable mode you must check potential targets, e.g. `CardUtil.getAllPossibleTargets`